tests/test_src/test_nuketesting/_cli/test_cli.py

```diff
@@ -0,0 +1,168 @@
+        yield runner_mock
+
+
+@pytest.fixture(autouse=True)
+def sys_exit() -> MagicMock:
+    """Mock the sys.exit for the testrunner tests."""
+    with patch("sys.exit") as good_bye:
+        yield good_bye
+
+
+def test_pass_arguments_to_data_object() -> None:
+    """Test simple pass of arguments to dataclass."""
+    cli_testrunner = CliRunner()
+
+    with patch("nuketesting._cli.main.CLIRunArguments") as test_run_arguments_mock:
+        cli_testrunner.invoke(main, ["-n", "nuke_path"])
+
+    test_run_arguments_mock.assert_called_once_with(
+        nuke_executable="nuke_path",
+        test_directory=".",
+        config=None,
+        run_in_terminal_mode=True,
+        pytest_args=(),
+        runner_name=None,
+    )
+
+
+def test_pass_all_arguments_to_data_object() -> None:
+    """Test passing of all possible arguments to dataclass object."""
+    cli_testrunner = CliRunner()
+
+    with patch("nuketesting._cli.main.CLIRunArguments") as test_run_arguments_mock:
+        cli_testrunner.invoke(
+            main,
+            [
+                "-n",
+                "nuke_path",
+                "-t",
+                "test_dir",
+                "-c",
+                "config/path.json",
+                "--terminal",
+                "false",
+                "-p",
+                "-v test",
+                "-p",
+                "-x",
+                "-r",
+                "Boomer",
+            ],
+        )
+
+    test_run_arguments_mock.assert_called_once_with(
+        nuke_executable="nuke_path",
+        test_directory="test_dir",
+        config="config/path.json",
+        run_in_terminal_mode=False,
+        pytest_args=("-v test", "-x"),
+        runner_name="Boomer",
+    )
+
+
+def test_run_arguments_convert_to_path() -> None:
+    """Test that the CLIRunArguments converts the paths to pathlib."""
+    test_cli_run_arguments = CLIRunArguments("test", (), "executable", "config")
+
+    assert test_cli_run_arguments.test_directory == Path("test")
+    assert test_cli_run_arguments.nuke_executable == Path("executable")
+    assert test_cli_run_arguments.config == Path("config")
+
+
+def test_exception_when_not_enough_arguments() -> None:
+    """Test to raise a TestRunCommandError when neither exe or config is provided."""
+    with pytest.raises(CLICommandError, match="Neither a config or a Nuke executable is provided."):
+        CLIRunArguments(".")
+
+
+def test_logger_called_during_error() -> None:
+    with patch("nuketesting._cli.main.logger") as logger_mock:
+        cli_testrunner = CliRunner()
+        cli_testrunner.invoke(main, [])
+
+    logger_mock.error.assert_called_once_with("An error occured: 'Neither a config or a Nuke executable is provided.'")
+
+
+def test_runner_executed(runner: MagicMock) -> None:
+    """Test that the runner is executed."""
+    instance = runner.return_value
+    cli_testrunner = CliRunner()
+
+    cli_testrunner.invoke(main, ["-n nuke_path"])
```

Comment:
Are we missing a comma here?
```suggestion
    cli_testrunner.invoke(main, ["-n","nuke_path"])
```